BZ1833574: Added a new section having an example insecure route (#34812)
openshift/openshift-docs: modules/nw-creating-a-route.adoc

[id="creating-a-route"]
= Creating a route

> A route allows you to host your application at a public URL. It can either be secure or insecure depending on the network security configuration of your application. An insecure route uses the basic HTTP routing protocol and exposes a service on an insecure application port.

Reviewer: Add a comma- It can either be secure or insecure, depending on the network...

Author: Done. Thanks for the review.

> You can optionally define security, such as TLS, for a route. To know how to create a secure route, see https://docs.openshift.com/container-platform/4.4/networking/routes/secured-routes.html[secure routes].

Reviewer 2: Saying "security, such as TLS" suggests there are alternatives to using TLS, but TLS is the crux of security for routes.  Also, shouldn't we use relative links?  
```suggestion
You can optionally configure a route to use TLS for security. To know how to create a secure route, see xref:../../networking/routes/secured-routes.html[secure routes].
```

Author: @Miciah I have rephrased it.

The following procedure
describes how to create a simple HTTP-based route to an example web application, https://github.com/openshift/origin/tree/master/examples/hello-openshift[hello-openshift].

.Prerequisites

* You installed the `oc` command line interface.
* You are logged in as an administrator.

.Procedure

> . Create a project called `hello_openshift` by running the following command:
> +
> [source,terminal]
> ----
> $ oc new-project hello_openshift

Reviewer 2: Namespaces and host names should use hyphens rather than underscores:
```suggestion
. Create a project called `hello-openshift` by running the following command:
+
[source,terminal]
----
$ oc new-project hello-openshift
```

Author: @Miciah Addressed.

----
+
. Create a pod in the project by running the following command:
+
[source,terminal]
----
$ oc create -f https://raw.githubusercontent.com/openshift/origin/master/examples/hello-openshift/hello-pod.json
----
+
. Create a service called `hello-openshift` by running the following command:
+
[source,terminal]
----
$ oc expose pod/hello-openshift
----
+

> . Create an insecure route to the `hello-openshift` application by running the following command:

Reviewer 3: @sferich888 , I like the idea of having an example in both. :)

+
[source,terminal]
----
$ oc expose svc hello-openshift
----
+

.YAML definition of the created insecure route:
[source,yaml]
apiVersion: route.openshift.io/v1
kind: Route
metadata:
  name: hello-openshift
spec:

>   host: hello-openshift-hello_openshift.<Ingress_Domain> <1>

Reviewer 2: ```suggestion
  host: hello-openshift-hello-openshift.<Ingress_Domain> <1>
```

Author: @Miciah Addressed.

  port:
    targetPort: 8080
  to:
    kind: Service
    name: hello-openshift
+

> `<Ingress_Domain>` is the default ingress domain name.

Reviewer: Is this meant to be a code callout?

Author: Yes; changed it to a callout.
